table/table_builder.h

```diff
@@ -77,6 +80,8 @@ struct TableReaderOptions {
   // largest seqno in the table
   SequenceNumber largest_seqno;
   BlockCacheTracer* const block_cache_tracer;
+  // memtable size limit at time of reader creation (can be zero when unknown).
+  const size_t write_buffer_size;
```

Comment:
It's something we can consider whether it is possible to prevent. If understanding the component of table reader needs some knowledge to "write buffer", it is less maintainable. Is it possible to rename it to something like "max_file_size_for_l0_index_pin" or something like that?

Reply:
Done.